File: frontend/src/src/actions/sms_api.ts
import { gql } from "@apollo/client";

export const FILTER_SMS_API = gql`
  query GetSMSApis(
    $filters: [Filter!]
    $orders: [Order!]
    $pagination: Pagination
  ) {
    response: getSMSApis(
      filters: $filters
      orders: $orders
      pagination: $pagination
    ) {
      smsApis {
        id
        name
        url
        agent
        password
        token
        status
        orderNum
        createdAt
        updatedAt
        deletedAt
      }
      total
    }
  }
`;

export const CREATE_SMS_API = gql`
  mutation NewSMSApi($input: NewSMSApiInput!) {
    response: createSMSApi(input: $input) {
      id
      name
      url
      agent
      password
      token
      status
      orderNum
      createdAt
      updatedAt
      deletedAt
    }
  }
`;

export const UPDATE_SMS_API = gql`
  mutation UpdateSMSApi($id: ID!, $input: UpdateSMSApiInput!) {
    response: updateSMSApi(id: $id, input: $input) {
      id
      name
      url
      agent
      password
      token
      status
      orderNum
      createdAt
      updatedAt
      deletedAt
    }
  }
`;

export const DELETE_SMS_API = gql`
  mutation DeleteSMSApi($id: ID!) {
    response: deleteSMSApi(id: $id)
  }
`;

File: backend/src/internal/models/sms_api.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type SMSApi struct {
	ID uint `json:"id"`

	Name     string `gorm:"size:500;" json:"name"`
	Url      string `gorm:"size:500;" json:"url"`
	Agent    string `gorm:"size:500;" json:"agent"`
	Password string `gorm:"size:500;" json:"password"`
	Token    string `gorm:"size:500;" json:"token"`

	OrderNum uint `json:"orderNum" gorm:"default:1"`

	Status bool `json:"status"`

	CreatedAt time.Time       `json:"createdAt"`
	UpdatedAt time.Time       `json:"updatedAt"`
	DeletedAt *gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}
